test/utils/document.js

```diff
@@ -0,0 +1,41 @@
+/* global window */
+
+export default class DocumentTest {
+  // Simplify the testing of deck.gl work that requires a document and canvas
+  constructor(canvasCallback) {
+    this.document = this._createDocument();
+    this.canvasParent = null;
+    this.testObject = null;
+    this._setup(canvasCallback);
+  }
+
+  _createDocument() {
+    // Creates a global document object via JSDOM or a deep clone of the browser one if available
+    if (typeof window === undefined || !window.document) {
+      const {JSDOM} = require('jsdom');
+      const dom = new JSDOM(`<!DOCTYPE html>`);
+      return dom.window.document;
+    }
+    const documentClone = window.document.cloneNode(true);
+    return documentClone;
+  }
+
+  _setup(canvasCallback) {
+    // Creates a canvas which can be passed to canvasCallback
+    // to add a node to
+    const canvas = this.document.createElement('canvas');
+
+    const canvasParent = this.document.createElement('div');
+    canvasParent.className = 'canvas-parent';
+    canvasParent.appendChild(canvas);
+    this.canvasParent = canvasParent;
+
+    this.document.body.appendChild(canvasParent);
+    this.testObject = canvasCallback(canvas);
+  }
+
+  finalize() {
+    this.testObject.remove();
```

Comment:
This class has no control over the returned value of the user callback. Assuming `testObject` is a DOM element here is dangerous. There is no apparent reason why `testObject` has to be stored in this class. It will be a lot easier to read if you write your test like this

```js
const testCanvas = new TestCanvas();
const myElement = setup(testCanvas.canvas);
// ...
myElement.remove();
testCanvas.finalize();
```

Reply:
Just updated the PR, the class has been cut in favor of smaller functions and an API that looks more like the one here.